extensions/ql-vscode/src/extension.ts

```diff
@@ -438,6 +440,27 @@ async function activateWithInstalledDistribution(
       async (uri: Uri | undefined) => await compileAndRunQuery(false, uri)
     )
   );
+  ctx.subscriptions.push(
+    commands.registerCommand(
+      'codeQL.runQueries',
+      async (_: Uri | undefined, multi: Uri[]) => {
+        const [files, dirFound] = await gatherQlFiles(multi.map(uri => uri.fsPath));
+        // warn user and display selected files when a directory is selected because some ql
+        // files may be hidden from the user.
+        if (dirFound) {
+          const fileString = files.map(file => path.basename(file)).join(', ');
```

Comment:
We need to cut this off at some reasonable length --- if someone opens up the vscode-codeql starter workspace and simply selects the `javascript` query directory and does 'run queries' it makes a huge dialog box pushing the Ok button off the screen.

Reply:
I can put an arbitrary maximum number of queries to select, let's say 10. I could even add this as a config option, but let's hold off on that.